src/System.Memory/ref/System.Memory.cs

```diff
@@ -112,6 +112,24 @@ public static partial class MemoryExtensions
 }
 namespace System.Buffers
 {
+    public sealed partial class ArrayBufferWriter : System.Buffers.IBufferWriter<byte>, System.IDisposable
+    {
+        public ArrayBufferWriter() { }
+        public ArrayBufferWriter(int initialCapacity) { }
+        public int BytesAvailable { get { throw null; } }
+        public int BytesWritten { get { throw null; } }
+        public int Capacity { get { throw null; } }
```

Comment:
BytesAvailable + BytesWritten == Capacity? Do we need BytesAvailable? What does one do with it? Why do two have the Bytes prefix and the third doesn't?

Reply:
> BytesAvailable + BytesWritten == Capacity?

Yes.

> Do we need BytesAvailable? What does one do with it?

BytesAvailable is useful when the user wants to avoid resizing when they call GetSpan() since they can know up front if the data they want to write will fit or not in the leftover space.